x-pack/plugins/actions/server/actions_client/actions_client.ts

```diff
@@ -672,52 +659,7 @@ export class ActionsClient {
   }: Omit<ExecuteOptions, 'request' | 'actionExecutionId'>): Promise<
     ActionTypeExecutorResult<unknown>
   > {
-    const log = this.context.logger;
-
-    if (
-      (await getAuthorizationModeBySource(this.context.unsecuredSavedObjectsClient, source)) ===
-      AuthorizationMode.RBAC
-    ) {
-      const additionalPrivileges = this.getSystemActionKibanaPrivileges(actionId, params);
-      let actionTypeId: string | undefined;
-
-      try {
-        if (this.isPreconfigured(actionId) || this.isSystemAction(actionId)) {
-          const connector = this.context.inMemoryConnectors.find(
-            (inMemoryConnector) => inMemoryConnector.id === actionId
-          );
-
-          actionTypeId = connector?.actionTypeId;
-        } else {
-          // TODO: Optimize so we don't do another get on top of getAuthorizationModeBySource and within the actionExecutor.execute
-          const { attributes } = await this.context.unsecuredSavedObjectsClient.get<RawAction>(
-            'action',
-            actionId
-          );
-
-          actionTypeId = attributes.actionTypeId;
-        }
-      } catch (err) {
-        log.debug(`Failed to retrieve actionTypeId for action [${actionId}]`, err);
-      }
-
-      await this.context.authorization.ensureAuthorized({
-        operation: 'execute',
-        additionalPrivileges,
-        actionTypeId,
-      });
-    } else {
-      trackLegacyRBACExemption('execute', this.context.usageCounter);
-    }
-
-    return this.context.actionExecutor.execute({
-      actionId,
-      params,
-      source,
-      request: this.context.request,
-      relatedSavedObjects,
-      actionExecutionId: uuidv4(),
-    });
+    return execute({ context: this.context, actionId, params, source, relatedSavedObjects });
```

Comment:
What do you think of creating a type for `type ConnectorExecuteParams = Omit<ExecuteOptions, 'request' | 'actionExecutionId'>` and then making the `execute` function be called like `execute(this.context, connectorExecuteParams)` instead of creating a brand new def of `ConnectorExecuteParams`?

Reply:
Sure! Resolved in this commit [9e6c4bb](https://github.com/elastic/kibana/pull/194481/commits/9e6c4bb1deeac1c8e939a719e03060436e719cd9)